pkg/webhook/virtualmachine/virtualmachine.go

```diff
@@ -204,13 +204,46 @@ func (a *virtualMachineAnnotator) mutateUpdateVirtualMachinesFn(virtualMachine *
 	return nil
 }
 
-// isVirtualMachineSpecChanged checks if the vm spec changed in this webhook update request.
-// we want to update the timestamp on every change, but not on metadata changes, as they change all the time,
-// which will cause a unneeded Timestamp update.
+func isVirtualMachineChanged(previousVirtualMachine, virtualMachine *kubevirt.VirtualMachine) bool {
+	if isVirtualMachineSpecChanged(previousVirtualMachine, virtualMachine) {
+		return true
+	}
+	if isVirtualMachineMetadataChanged(previousVirtualMachine, virtualMachine) {
+		return true
+	}
+	return false
+}
+
+// isVirtualMachineChanged checks if the vm spec changed in this webhook update request.
 func isVirtualMachineSpecChanged(previousVirtualMachine, virtualMachine *kubevirt.VirtualMachine) bool {
 	return !reflect.DeepEqual(previousVirtualMachine.Spec, virtualMachine.Spec)
 }
 
+// isVirtualMachineMetadataChanged checks if non-automatically generated metadata fields changed in this webhook
+// update request.
+func isVirtualMachineMetadataChanged(previousVirtualMachine, virtualMachine *kubevirt.VirtualMachine) bool {
+	if previousVirtualMachine.GetNamespace() != virtualMachine.GetNamespace() {
```

Comment:
Correct me if I am wrong, if the exact same VM will be create on different namespace (using the same yaml file but create it on a different namespace) it will be recognized as VM to update right?
I think we should teat such VM as another VM in the cluster.

Reply:
Speaking of namespace, how could a namespace of a VM change? AFAIK namespace is not mutable for any k8s resource

Reply:
Not sure I follow.

Reply:
ahh I think I get you ... a valid point. I'll remove this check then

Reply:
Done